### PR TITLE
Get trial directly from feature discovery

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Get trial directly from feature discovery

#### Ticket Link
[MM-26412](https://mattermost.atlassian.net/browse/MM-26412)

### CODE REVIEW

**Comment by jespino** [comment]
@michaelgamble that comment doesn't affect to this PR (Actually that are text changes related to exactly that here: https://mattermost.atlassian.net/browse/MM-26481) by the way, I going to "transplant" that changes into this screens too. So, at the bottom of this, you will see 2 check boxes and the "Alternatively". But before all that happens, we need to merge this PR.

**Comment by stevemudie** [comment]
Am I misunderstanding the ticket? Shouldn't the text on the buttons read "Start a Trial" as well?
Start a trial

**Comment by jespino** [comment]
@stevemudie probably, this is 2 different features getting merged in one, so @michaelgamble is looking into this.

**Comment by esadur** [comment]
I think there might have been a misunderstanding. The idea was that if you click the button on the AD/LDAP or SAML feature discovery (disco) pages, it takes you to the Edition and License page (instead of the webpage), and you start the trial from there. Wasn't intending to have the full language and trial initiation on each disco page.

The idea was to leave the disco pages as they were (just changing the "Request Trial" button to "Start a Trial" to match the language), and then have the Edition and License page do the heavy lifting.

File: components/admin_console/feature_discovery/ldap.tsx [hunk 1]
```diff
@@ -15,8 +15,7 @@ const LDAPFeatureDiscovery: React.FC = () => {
             titleDefault='Integrate Active Directory / LDAP with Mattermost Enterprise E10'
             copyID='admin.ldap_feature_discovery.copy'
             copyDefault={'When you connect Mattermost with your organization\'s Active Directory / LDAP, users can log in without having to create new usernames and passwords.'}
```
Comment (justinegeffen) [inline components/admin_console/feature_discovery/ldap.tsx:17]:
```suggestion
            copyDefault={'When you connect Mattermost with your organization\'s Active Directory/LDAP, users can log in without having to create new usernames and passwords.'}
```

**Comment by justinegeffen** [comment]
> I would change the content from "Or get a trial license manually at https://mattermost.com/trial" to "Alternatively, get a trial license manually at https://mattermost.com/trial"
> 
> It's not great practice for a sentence to seemingly start with the word "Or" IMO @justinegeffen do you agree?

I'm +1 on "Alternatively" and have read @jespino's comment - and also added my 2c to https://mattermost.atlassian.net/browse/MM-26481.

File: components/admin_console/feature_discovery/feature_discovery.tsx [hunk 2]
```diff
@@ -13,68 +20,123 @@ type Props = {
...
+                        />
+                    </div>
+                    <a
+                        className='btn'
+                        data-testid='featureDiscovery_primaryCallToAction'
+                        onClick={this.requestLicense}
+                    >
+                        <LoadingWrapper
+                            loading={this.state.gettingTrial}
+                            text={Utils.localizeMessage('admin.license.trial-request.loading', 'Getting trial')}
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/admin_console/feature_discovery/feature_discovery.tsx:123]: That is a very long text, and breaks the design in the feature discovery pages.

**Comment by jespino** [comment]
@michaelgamble @esadur please, take a look to the last changes and comment or approve the PR.

**Comment by jespino** [comment]
@esadur done :)

**Comment by jespino** [comment]
Last proposed changes done, please @michaelgamble and @esadur review this ASAP.

**Comment by andrewbrown00** [comment]
@jespino did the changes make it on https://mattermost-webapp-pr-5779.test.mattermost.cloud/? I still see the quotes...

image

**Comment by jespino** [comment]
That is because the spinmint server hasn't been updated. Lets destroy and create it again.

**Comment by justinegeffen** [comment]
> @jespino did the changes make it on https://mattermost-webapp-pr-5779.test.mattermost.cloud/? I still see the quotes...
> 
> image

It's in bold now, but both buttons say "Start trial" and the wording says **Start a trial**.

File: components/admin_console/feature_discovery/feature_discovery.tsx [hunk 3]
```diff
@@ -4,77 +4,155 @@
...
+                        <FormattedMessage
+                            id='admin.ldap_feature_discovery.call_to_action.secondary'
+                            defaultMessage='Learn more'
+                        />
+                    </a>
+                    {gettingTrialError}
+                    <p className='trial-legal-terms'>
+                        <FormattedMarkdownMessage
+                            id='admin.license.trial-request.accept-terms'
+                            defaultMessage='By clicking "Start a trial," I agree to the [Mattermost Software Evaluation Agreement](!https://mattermost.com/software-evaluation-agreement/), [Privacy Policy](!https://mattermost.com/privacy-policy/), and receiving product emails.'
```
Comment (esadur) [inline components/admin_console/feature_discovery/feature_discovery.tsx]: Change `Start a trial` to `Start trial`

File: components/admin_console/license_settings/license_settings.jsx [hunk 4]
```diff
@@ -211,24 +221,17 @@ export default class LicenseSettings extends React.PureComponent {
...
                     </p>
-                    <p className='help-text'>
-                        <FormattedMessage
-                            id='admin.license.trial-request.different-data'
-                            defaultMessage='Or get a trial license manually at '
+                    {gettingTrialError}
+                    <p className='trial-legal-terms'>
+                        <FormattedMarkdownMessage
+                            id='admin.license.trial-request.accept-terms'
+                            defaultMessage='By clicking "Start a trial," I agree to the [Mattermost Software Evaluation Agreement](!https://mattermost.com/software-evaluation-agreement/), [Privacy Policy](!https://mattermost.com/privacy-policy/), and receiving product emails.'
```
Comment (esadur) [inline components/admin_console/license_settings/license_settings.jsx]: Change `Start a trial` to `Start trial`

File: i18n/en.json [hunk 5]
```diff
@@ -951,9 +951,10 @@
   "admin.license.noFile": "No file uploaded",
   "admin.license.removing": "Removing License...",
   "admin.license.title": "Edition and License",
-  "admin.license.trial-request.different-data": "Or get a trial license manually at ",
+  "admin.license.trial-request.accept-terms": "By clicking \"Start a trial,\" I agree to the [Mattermost Software Evaluation Agreement](!https://mattermost.com/software-evaluation-agreement/), [Privacy Policy](!https://mattermost.com/privacy-policy/), and receiving product emails.",
```
Comment (esadur) [inline i18n/en.json]: Change `Start a trial` to `Start trial`

**Comment by cpanato** [comment]
removing the test server because everybody already review that and waiting to be merged

**Pull request merged**